docs/iam.rst

```diff
@@ -0,0 +1,104 @@
+Using IAM For Users
+===================
+
+This document does not explain how Google Cloud IAM works, for that, please check out `the docs <https://cloud.google.com/iam/docs/>`_.
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Methods
+----------------------
+
+>>> resource.add_roles(iam.user('alice@example.com'), resource.roles.OWNER, resource.roles.EDITOR)
+True
+
+The first argument is one of the IAM "Member" types:
+
+- ``iam.user(email)`` an individual Google account
+- ``iam.service_account(email)`` a Google Cloud Service Account
+- ``iam.group(email)`` a Google group.
+- ``iam.domain(domain_name)`` a Google-For-Work domain
+- ``iam.ALL_AUTHENTICATED_USERS`` any authenticated user
+- ``iam.ALL_USERS``
+
+Note that all of these are convenience wrappers around strings. See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+
+All following arguments are roles to added to the specified member. Roles are simply strings that correspond to sets of permissions.
+The resource should provide an enumeration of the possible "curated roles" for the resource.
+You can acquire this list by running ``gcloud iam list-grantable-roles //fully/qualified/resource/path``.
+
+This method returns ``True`` if the change to the resource's policy was made successfully, and returns ``False`` otherwise
+(such as in the case of an update being made during this update).
+
+>>> resource.remove_roles(iam.group('devs@example.com'), resource.roles.OWNER, resource.roles.EDITOR)
+False
+
+Same as above, but removes the specified roles from the specified member.
+
+>>> resource.add_members(resource.roles.OWNER, iam.domain('example.com'), iam.service_account('compute@iam.my-project.example.com'))
+True
+
+Same as above, but the first argument is a role, and all subsequent arguments are members to which that role should be added.
+
+>>> resource.remove_members(resource.roles.OWNER, iam.ALL_USERS)
+True
+
+Same as above, but removes all listed members from the specified role.
+
+
+>>> user_emails = ['alice@example.com']
+>>> def remove_fn(member):
+>>>     return iam.is_group(member) or member == iam.user('bob@example.com')
+>>> policy_change1 = iam.PolicyChange(resource.roles.OWNER).remove(iam.group('devs@example.com'))
+>>> for user in user_emails:
+>>>     policy_change1.add(iam.user(user))
+>>> policy_change2 = iam.PolicyChange(resource.roles.EDITOR).remove(remove_fn=remove_fn)
+>>> resource.update_policy(policy_change1, policy_change2)
```

Comment:
FYI still deeply unhappy with this. Have some ideas, which I'll propose in a followup commit.